sideboard/lib/_threads.py

```diff
@@ -12,24 +16,57 @@
 from sideboard.lib import log, on_startup, on_shutdown
 
 
-# inject our own code at the start of every thread's start() method which sets the thread name via prctl().
-# Python thread names will now be shown in external system tools like 'top', '/proc', etc.
-def _thread_name_insert(self):
-    if self.name:
+import threading
+
+
+def _get_linux_thread_tid():
+    """
+    Get the current linux thread ID as it appears in /proc/[pid]/task/[tid]
+    :return: Linux thread ID if available, or -1 if any errors / not on linux
+    """
+    try:
+        if not platform.system().startswith('Linux'):
+            raise ValueError
```

Comment:
Not that I expect this to ever actually come up, but in case it ever did, having an actual message with the error would be great, e.g.

```
raise ValueError('Can only get thread id on Linux systems')
```